pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -707,8 +707,8 @@ func (ns *nodeServer) mount(sourcePath, targetPath string, mountOptions []string
 // given id and returns the device manager which creates that volume.
 func (ns *nodeServer) getDeviceManagerForVolume(id string) (pmdmanager.PmemDeviceManager, error) {
 
-	vol, ok := ns.cs.pmemVolumes[id]
-	if !ok {
+	vol := ns.cs.getVolumeByID(id)
+	if vol == nil {
```

Comment:
As a quick fix this is okay. A better long-term fix would be to arrange the code so that `pmemVolumes` is not accessible directly without locking, to prevent making the same mistake again.